sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/models/TextAnalyticsSubscriptionKeyCredential.java

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.ai.textanalytics.models;
+
+/**
+ * Subscription key credential that shared across cognitive services, or restrict to single service.
+ */
+public final class TextAnalyticsSubscriptionKeyCredential {
+    private String subscriptionKey;
+
+    /**
+     * Creates a {@code SharedKeyCredential} model that describes subscription key.
+     *
+     * @param subscriptionKey the subscription key
+     */
+    public TextAnalyticsSubscriptionKeyCredential(String subscriptionKey) {
+        this.subscriptionKey = subscriptionKey;
+    }
+
+    /**
+     * Get the subscription key value.
+     *
+     * @return the subscription key value
+     */
+    public String getSubscriptionKey() {
+        return this.subscriptionKey;
+    }
+
+    /**
+     * Set the subscription key value.
+     *
+     * @param subscriptionKey the subscription key
+     * @return the {@link TextAnalyticsSubscriptionKeyCredential} itself
+     */
+    public TextAnalyticsSubscriptionKeyCredential updateCredential(String subscriptionKey) {
+        this.subscriptionKey = subscriptionKey;
```

Comment:
Is this thread safe? Is there a way to guarantee that when modified it updates everywhere?

Reply:
I had a few discussions on the thread-safe concern. some believe it is thread-safe. the reason is it is a single atomic assignment. But depends on how you use it, It could be not threaded safe. This is one I am not sure yet. I will give you an update once finalized.

Reply:
sure, see here: https://docs.microsoft.com/en-us/azure/cognitive-services/authentication#authenticate-with-a-multi-service-subscription-key

Reply:
In the pipeline policy, it is fetched, so it's possible that the policy picks up a different value than the one that is set. This is not thread safe.